src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9267,24 +9134,29 @@ private ForStatementSyntax ParseForStatement(SyntaxList<AttributeListSyntax> att
                     this.ParseForStatementExpressionList(ref semi2, incrementors);
                 }
 
-                var closeParen = this.EatToken(SyntaxKind.CloseParenToken);
-                var statement = ParseEmbeddedStatement();
-                return _syntaxFactory.ForStatement(attributes, forToken, openParen, decl, initializers, semi, condition, semi2, incrementors, closeParen, statement);
+                return _syntaxFactory.ForStatement(
+                    attributes,
+                    forToken,
+                    openParen,
+                    decl,
+                    _pool.ToListAndFree(initializers),
+                    semi,
+                    condition,
+                    semi2,
+                    _pool.ToListAndFree(incrementors),
+                    this.EatToken(SyntaxKind.CloseParenToken),
+                    ParseEmbeddedStatement());
             }
             finally
             {
                 _termState = saveTerm;
                 this.Release(ref resetPoint);
```

Comment:
i want to make it possible to use a simple `using` statement for reset-points in the future.  Needing a try/finally to simulate this feels silly.